src/endless/EndlessUsbTool.cpp

```diff
@@ -195,31 +195,40 @@ void CEndlessUsbToolApp::InitLogging()
 	fileName.Replace(L" ", L"");
 	fileName += s;
 	fileName += L".log";
-	if (CEndlessUsbToolDlg::IsUninstaller()) {
-		fileName = TempFilePath(fileName);
-	} else {
-		fileName = GET_LOCAL_PATH(fileName);
-	}
-
-	try {
-		BOOL result = m_logFile.Open(fileName, CFile::modeWrite | CFile::typeUnicode | CFile::shareDenyWrite | CFile::modeCreate | CFile::osWriteThrough);
-		if (!result) {
-			m_enableLogDebugging = false;
-		}
 
-		//uprintf("Log original date time %ls\n", s);
-		//uprintf("Application version: %s\n", RELEASE_VER_STR);
-		//uprintf("Windows version: %s\n", WindowsVersionStr);
-		//uprintf("Windows version number: 0x%X\n", nWindowsVersion);
-		//uprintf("-----------------------------------\n", s);
+	BOOL result = false;
+	// Try to write log alongside ourself, provided we're not in C:\endless
+	// which is unwriteable and about to be deleted(!)
+	if (!CEndlessUsbToolDlg::IsUninstaller()) {
```

Comment:
`IsUninstaller()` is true iff the executable is called `endless-uninstaller.exe`. In the general case of the app switching itself into uninstaller mode when `C:\endless\endless.img` exists, the logfile will be written in the current dir.

Happy to remove this guard if you prefer (the effect should be the same: creating the file will fail and we'll fall back to the tmpdir), though.